routemaster/state_machine/api.py

```diff
@@ -178,7 +190,13 @@ def _process_transitions_for_metadata_update(
                 "Label not in a gate",
             )
 
-        could_progress = process_gate(app, current_state, label, conn)
+        could_progress = process_gate(
+            app,
+            current_state,
+            state_machine,
+            label,
+            conn,
+        )
```

Comment:
This function has reached the point where I feel like it should be kwargs-only.

Reply:
48de4dca8a3811fb867363d665eb790a4ac38427